ckanext/datastore/cli.py

```diff
@@ -111,3 +112,48 @@ def _parse_db_config(config_key=u'sqlalchemy.url'):
         )
         raise click.Abort()
     return db_config
+
+
+@datastore.command(
+    u'purge',
+    short_help=u'purge orphaned resources from the datastore.'
+)
+def purge():
+    u'''Purge orphaned resources from the datastore using the datastore_delete
+    action, which drops tables when called without filters.'''
+
+    site_user = logic.get_action(u'get_site_user')({u'ignore_auth': True}, {})
+    context = {u'user': site_user[u'name']}
+
+    result = logic.get_action(u'datastore_search')(
+        context,
+        {u'resource_id': u'_table_metadata'}
+    )
+
+    resource_id_list = []
+    for record in result[u'records']:
+        try:
+            logic.get_action(u'resource_show')(
+                context,
+                {u'id': record[u'name']}
+            )
+            click.echo(u"Resource '%s' found" % record[u'name'])
+        except logic.NotFound:
+            resource_id_list.append(record[u'name'])
+            click.echo(u"Resource '%s' orphaned - queued for drop" %
+                       record[u'name'])
+        except KeyError:
+            continue
+
+    # Drop the orphaned datastore tables. When datastore_delete is called
+    # without filters, it does a drop table cascade
```

Comment:
Does this mean it deletes the record on `_table_metadata`?

Reply:
Effectively yes.  As `_table_metadata` is a view of all the existing tables in the datastore.

https://github.com/ckan/ckan/blob/5156a4024a1aea7af6ad927b4af8a70aab8597ee/ckanext/datastore/set_permissions.sql#L53-L71

So dropping a resource table removes the corresponding record on `_table_metadata`.